### ISSUE TITLE
Option to exclude all `model_extras` from `repr`

### ISSUE THREAD

**Issue by it176131** [pydantic/pydantic#9732]
### Initial Checks

- [X] I have searched Google & GitHub for similar requests and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this feature is missing

### Description

The [`model_config`](https://docs.pydantic.dev/latest/api/base_model/#pydantic.BaseModel.model_config) attribute gives us the ability to `"allow"`, `"ignore"`, or `"forbid"` extra values, but I'd like to hide them similar to how `Field(..., repr=False)` can hide required/expected/known fields. My current approach overrides the models [`__repr_args__`](https://github.com/pydantic/pydantic/blob/c08dacdebc3a35a1a76ffbded593cab583e9f951/pydantic/main.py#L1024) method and excludes the logic around `pydantic_extra`:

```diff
    def __repr_args__(self) -> _repr.ReprArgs:
        for k, v in self.__dict__.items():
            field = self.model_fields.get(k)
            if field and field.repr:
                yield k, v

-       # `__pydantic_extra__` can fail to be set if the model is not yet fully initialized.
-       # This can happen if a `ValidationError` is raised during initialization and the instance's
-       # repr is generated as part of the exception handling. Therefore, we use `getattr` here
-       # with a fallback, even though the type hints indicate the attribute will always be present.
-       try:
-           pydantic_extra = object.__getattribute__(self, '__pydantic_extra__')
-       except AttributeError:
-           pydantic_extra = None
-
-       if pydantic_extra is not None:
-           yield from ((k, v) for k, v in pydantic_extra.items())
        yield from ((k, getattr(self, k)) for k, v in self.model_computed_fields.items() if v.repr)
```

But this seems hacky. I'd like to propose a new setting for the `model_config`, `"extra_repr"`, and wrap the `pydantic_extra` logic in `__repr_args__` with a conditional:

```diff
    def __repr_args__(self) -> _repr.ReprArgs:
        for k, v in self.__dict__.items():
            field = self.model_fields.get(k)
            if field and field.repr:
                yield k, v

+       if self.model_config.get("extra_repr", True):
+           # `__pydantic_extra__` can fail to be set if the model is not yet fully initialized.
+           # This can happen if a `ValidationError` is raised during initialization and the instance's
+           # repr is generated as part of the exception handling. Therefore, we use `getattr` here
+           # with a fallback, even though the type hints indicate the attribute will always be present.
+           try:
+               pydantic_extra = object.__getattribute__(self, '__pydantic_extra__')
+           except AttributeError:
+               pydantic_extra = None

+           if pydantic_extra is not None:
+               yield from ((k, v) for k, v in pydantic_extra.items())
+
        yield from ((k, getattr(self, k)) for k, v in self.model_computed_fields.items() if v.repr)
```

Minimal example with expected output _after_ proposed changes:
```python
from pydantic.config import ConfigDict
from pydantic.main import BaseModel


class Foo(BaseModel):

    model_config = ConfigDict(extra="allow", extra_repr=False)

    name: str


foo = Foo(name="Bar", Baz="Spam", Eggs="Spam", More="Spam")
print(repr(foo))
# >>> Foo(name='Bar')

print(foo.model_extra)  # No effect.
# >>> {'Baz': 'Spam', 'Eggs': 'Spam', 'More': 'Spam'}

print(foo.model_dump())  # No effect.
# >>> {'name': 'Bar', 'Baz': 'Spam', 'Eggs': 'Spam', 'More': 'Spam'}

```

As you can see, this keeps the `Foo` representation clean/succinct without removing the `model_extra`.

### Affected Components

- [ ] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [ ] [Data validation/parsing](https://docs.pydantic.dev/concepts/models/#basic-model-usage)
- [ ] [Data serialization](https://docs.pydantic.dev/concepts/serialization/) - `.model_dump()` and `.model_dump_json()`
- [ ] [JSON Schema](https://docs.pydantic.dev/concepts/json_schema/)
- [ ] [Dataclasses](https://docs.pydantic.dev/concepts/dataclasses/)
- [ ] [Model Config](https://docs.pydantic.dev/concepts/config/)
- [ ] [Field Types](https://docs.pydantic.dev/api/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://docs.pydantic.dev/concepts/validation_decorator/)
- [ ] [Generic Models](https://docs.pydantic.dev/concepts/models/#generic-models)
- [X] [Other Model behaviour](https://docs.pydantic.dev/concepts/models/) - `model_construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by sydney-runkle** [comment]
Hi @it176131,

Thanks for the feature request.

This reminds me a bit of https://github.com/pydantic/pydantic/issues/9415.

On one hand, I think having a `config` setting that makes `repr` modification easier would be helpful - especially for some common cases related to `extra`, defaults, etc. 

At the same time, I haven't seen too much demand for this, and I think the approach where you override the `__repr_args__` function makes sense for your case in the short term.

Curious to hear other ideas from folks re what API might make the most sense in `model_config` going forward.

**Comment by sydney-runkle** [comment]
I'll also note, I think we can close either this issue or the other one + merge them into one feature request once we've ironed out some of the details.

**Comment by it176131** [comment]
> Hi @it176131,
> 
> Thanks for the feature request.
> 
> This reminds me a bit of #9415.
> 
> On one hand, I think having a `config` setting that makes `repr` modification easier would be helpful - especially for some common cases related to `extra`, defaults, etc.
> 
> At the same time, I haven't seen too much demand for this, and I think the approach where you override the `__repr_args__` function makes sense for your case in the short term.
> 
> Curious to hear other ideas from folks re what API might make the most sense in `model_config` going forward.

After reading over #9415 again I do see some similarities. Combing over my searches, #1831, #9712, and #3192 pop out as well.


> I'll also note, I think we can close either this issue or the other one + merge them into one feature request once we've ironed out some of the details.

I'd be open to combining requests and collaborating on a PR.

**Comment by sydney-runkle** [comment]
@it176131,

Sounds great. Let me chat with the open source team tomorrow morning about potential APIs for this, then I'll get back to you with our thoughts :).

**Comment by sydney-runkle** [comment]
Yeah honestly, I think we'd lean more on the side of customizing `__repr_args__` in user code for now.

That being said, if you have a compelling proposal for a way to add support for this, I'd be happy to review a PR :).

**Comment by it176131** [comment]
> Yeah honestly, I think we'd lean more on the side of customizing `__repr_args__` in user code for now.
> 
> That being said, if you have a compelling proposal for a way to add support for this, I'd be happy to review a PR :).

Sure thing! I'll link back to this issue.

**Comment by it176131** [comment]
@sydney-runkle should we close this as not planned?
